### PR TITLE
Common locals for body template

### LINKED ISSUE #114: Common locals for body template
waiting for https://github.com/claranet/terraform-signalfx-detectors/issues/105 we could use a symlink with common locals.tf file to homogenize, simplify and put in common some info like body template.

### PR DESCRIPTION
**Pull request by xp-1000**
Originally aims to bring body template.

#114 has evolved to rationalize some common code using a common locals.tf file.

this should simplify code and enjoy reuse waiting for #105.

fix #114

### CODE REVIEW

**Review by Shr3ps — APPROVED** [review]

**Review by pdecat — APPROVED** [review]

File: common/locals.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
+locals {
+  heartbeat_filters_gcp   = "(not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}'))"
```
Comment (BzSpi) [inline common/locals.tf:2]:
```suggestion
  not_running_vm_filters_gcp   = "(not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}'))"
```
This filter may be used for other things than heartbeat.

File: common/locals.tf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  heartbeat_filters_aws   = "(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}'))"
+  heartbeat_filters_azure = "(not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated'))"
+  heartbeat_filters = format(
+    "%s and %s and %s",
+    local.heartbeat_filters_aws,
+    local.heartbeat_filters_gcp,
+    local.heartbeat_filters_azure
+  )
+  name_prefix     = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}]"
```
Comment (BzSpi) [inline common/locals.tf:11]:
```suggestion
  detector_name_prefix     = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}]"
```

File: common/locals.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  subject_prefix  = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}}"
```
Comment (BzSpi) [inline common/locals.tf:12]:
```suggestion
  alert_subject_prefix  = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}}"
```

File: common/locals.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  subject_suffix  = "on {{{dimensions}}}"
+  subject         = format("%s ({{inputs.signal.value}}) %s", local.subject_prefix, local.subject_suffix)
+  subject_novalue = format("%s %s", local.subject_prefix, local.subject_suffix)
+  body            = <<-EOF
```
Comment (BzSpi) [inline common/locals.tf:16]:
```suggestion
  alert_body            = <<-EOF
```

**Review by BzSpi — CHANGES_REQUESTED** [review]

**Review by BzSpi — APPROVED** [review]
:+1:

**Pull request merged**